x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/InitialNodeSecurityAutoConfiguration.java

```diff
@@ -88,81 +93,103 @@ public static void maybeGenerateEnrollmentTokensAndElasticCredentialsOnNodeStart
-                // time a node starts it will form a cluster by itself and can hold the .security index (which we assume it is when
-                // {@code ENROLLMENT_ENABLED} is true), that the node process's output is a terminal and that the password is not
-                // specified already via the two secure settings
-                if (false == BOOTSTRAP_ELASTIC_PASSWORD.exists(environment.settings())
-                    && false == AUTOCONFIG_ELASTIC_PASSWORD_HASH.exists(environment.settings())) {
-                    final char[] elasticPassword = generatePassword(20);
-                    nativeUsersStore.createElasticUser(elasticPassword, ActionListener.wrap(aVoid -> {
-                        LOGGER.debug("elastic credentials generated successfully");
-                        groupedActionListener.onResponse(Map.of("generated_elastic_user_password", new String(elasticPassword)));
-                    }, e -> {
-                        LOGGER.error("Failed to generate credentials for the elastic built-in superuser", e);
-                        // null password in case of error
-                        groupedActionListener.onResponse(Map.of());
-                    }));
-                } else {
-                    if (false == BOOTSTRAP_ELASTIC_PASSWORD.exists(environment.settings())) {
-                        LOGGER.info(
-                            "Auto-configuration will not generate a password for the elastic built-in superuser, "
-                                + "you should use the password specified in the node's secure setting ["
-                                + BOOTSTRAP_ELASTIC_PASSWORD.getKey()
-                                + "] in order to authenticate as elastic"
-                        );
+                threadPool.generic().execute(new AbstractRunnable() {
+
+                    @Override
+                    public void onFailure(Exception e) {
+                        LOGGER.error("Unexpected exception when auto configuring the initial node for Security", e);
                     }
-                    // empty password in case password generation is skipped
-                    groupedActionListener.onResponse(Map.of("generated_elastic_user_password", ""));
-                }
-                final Iterator<TimeValue> backoff = BACKOFF_POLICY.iterator();
-                enrollmentTokenGenerator.createKibanaEnrollmentToken(kibanaToken -> {
-                    if (kibanaToken != null) {
+
+                    @Override
+                    protected void doRun() throws Exception {
+                        // the HTTP address is guaranteed to be bound only after the node started
+                        nodeStartedSignal.await();
+                        // a lot of stuff runs when a node just started, and the autoconfiguration is not time-critical
+                        // and nothing else depends on it; be a good sport and wait a couple
+                        Thread.sleep(9000);
```

Comment:
Is there are reason for `sleep` rather than a scheduled thread?
It's minor issue but it's better not to use up one on the generic threads sleeping.

Reply:
I overlooked the existence of the `threadpool.schedule` method.
I have pushed e0a33f329640fa63da3b .

Reply:
In the above I actually forgot to remove the sleep, I've later pushed 459d6a9 .